apps/native/app/src/screens/family/family-details.tsx

```diff
@@ -6,11 +6,40 @@ import {
   Navigation,
   NavigationFunctionComponent,
 } from 'react-native-navigation'
-import { useNationalRegistryChildrenQuery } from '../../graphql/types/schema'
+import {
+  NationalRegistryBioChildQuery,
+  NationalRegistryChildCustodyQuery,
+  NationalRegistrySpouseQuery,
+  useNationalRegistryBioChildQuery,
+  useNationalRegistryChildCustodyQuery,
+  useNationalRegistrySpouseQuery,
+} from '../../graphql/types/schema'
 import { createNavigationOptionHooks } from '../../hooks/create-navigation-option-hooks'
 import { formatNationalId } from '../../lib/format-national-id'
 import { testIDs } from '../../utils/test-ids'
 
+type ChildCustodyDetails = NonNullable<
+  NonNullable<
+    NonNullable<
+      NonNullable<NationalRegistryChildCustodyQuery['nationalRegistryPerson']>
+    >['childCustody']
+  >[number]
+>['details']
+
+type BioChildDetails = NonNullable<
+  NonNullable<
+    NonNullable<
+      NonNullable<NationalRegistryBioChildQuery['nationalRegistryPerson']>
+    >['biologicalChildren']
+  >[number]
+>['details']
+
+type SpouseDetails = NonNullable<
+  NonNullable<NationalRegistrySpouseQuery['nationalRegistryPerson']>
+>['spouse']
+
+type Person = ChildCustodyDetails | BioChildDetails | SpouseDetails
```

Comment:
Can't you use optional chaining instead of this nested waterfall of NonNullables. Something like
```
type ChildCustodyDetails = NonNullable<
  NationalRegistryChildCustodyQuery['nationalRegistryPerson']?.childCustody?.[number]?.details
>;

...
```

Not sure if this works but you should at least consider it for nicer readability

Reply:
No unfortunately, optional chaining does not work in type definitions from what I have tried.